main.py

```diff
@@ -16,6 +20,18 @@
     layout="wide"
 )
 
+session = None
+try: 
+    session = get_active_session()
+except Exception:
+    try:
+        session = Session.builder.configs(get_env_var_config()).create()
+    except Exception:
+        try:
+            session = st.connection.snowflake.login({'account': '', 'user': '', 'password': None, 'database': '', 'warehouse': ''})
+        except Exception as e:
+            raise e
+
```

Comment:
If user has a session available, use that. 
Else, create a session from the env vars
Else, prompt for login 
Else, throw exception